riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java

```diff
@@ -124,6 +123,13 @@ public void closeAsync() {
                         }
                     }
                 });
+                return;
+            }
+        }
+        synchronized (state) {
+            if (isValid()) {
```

Comment:
Why not putting this in `synchronized (this) { ... }`?

Reply:
Fixed. Based on the discussion, modified the code in NetworkClient to use synchronized (this) and remove synchronized (state).